app/main/authorisation.py

```diff
@@ -93,27 +73,57 @@ def get_auth(self, email: str) -> AuthBase | None:
         return auth
 
 
-def _auth_class_factory(fund: str) -> type[AuthBase]:
-    """Given a fund, returns the associated auth class.
+class ReadOnlyAuthMappings:
+    """Read-only dictionary wrapper for storing authorisation mappings."""
```

Comment:
> Why can't we use a dict of auth mappings for instance?
It's an option. I was using a dict of auth mapping to begin with along with a function that you pass the dictionary and a fund to, and it handles the retrieval from the dict. I think wrapping it in a class like this is beneficial because we remove all other dictionary functionality that we don't want and the code is also more self documenting - albeit with a bit of extra boilerplate. 

This container and it's contents should all be immutable - it only needs to be instantiated when the program starts up, and then we just want to read it's contents.

Reply:
Sounds good, think it's a good design choice to use this class which also specifies that the mappings are immutable